src/cmd/cli/command/compose.go

```diff
@@ -47,6 +61,7 @@ func makeComposeUpCmd() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			var force, _ = cmd.Flags().GetBool("force")
 			var detach, _ = cmd.Flags().GetBool("detach")
+			var waitTimeout, _ = cmd.Flags().GetInt("wait-timeout")
```

Comment:
I think you can replace the base context which is `cmd.Context()` with a context with timeout if there is a `wait-timeout` param. Something like:
```
ctx := cmd.Context()
if waitTimeout > 0 {
  ctx = context.WithTImeout(cmd.Context(), waitTimeout)
}
```
And use this ctx as the parent context for.

Also, I would suggest use `Duration` type for this flag, so we can use `--wait-timeout=3m`